### PR TITLE
[SYCL] Partial fix for stream problem

### PR DESCRIPTION
**Pull request by maximdimakov**
If work group size wasn't set explicitly then buffer for the stream object will be created with zero size. WG size is received later from work group size kernel attribute if set, or set to the default size with size = 1 for every dimension. Right now we can't get this attribute, it will be possible when the feature of containing kernel attributes by device image will be implemented.
So, when work group size is needed for stream, we suppose that its size is default for the correct working of stream in the cases when the attribute isn't set.
Test for this fix : https://github.com/intel/llvm-test-suite/pull/617
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

File: sycl/source/handler.cpp [hunk 1]
```diff
@@ -414,9 +414,21 @@ void handler::processArg(void *Ptr, const detail::kernel_param_kind_t &Kind,
         static_cast<detail::AccessorBaseHost *>(&S->GlobalFlushBuf);
     detail::AccessorImplPtr GFlushImpl = detail::getSyclObjImpl(*GFlushBase);
     detail::Requirement *GFlushReq = GFlushImpl.get();
+
+    auto GlobalSize = MNDRDesc.GlobalSize.size();
```
Comment (alexbatashev) [inline sycl/source/handler.cpp:418]: Nit. Please, use the correct type here.

**Review by alexbatashev — COMMENTED** [review]
LGTM

**Review by alexbatashev — APPROVED** [review]

**Comment by maximdimakov** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/617

**Comment by maximdimakov** [comment]
Failed tests are not related to this patch

**Comment by maximdimakov** [comment]
The test with applied patch is passed on all platforms. Fails on windows are not relied to this patch. Bug report is filed

**Comment by bader** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/617

**Comment by bader** [comment]
@maximdimakov, could you merge `sycl` branch to your `stream` branch, please? I'd like to get up-to-date validation results.

**Comment by bader** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/617

**Comment by maximdimakov** [comment]
@bader I merged `sycl` branch to `stream` branch.

**Comment by bader** [comment]
@tfzhu, @DoyleLi, do you know why validation status for https://github.com/intel/llvm-test-suite/pull/617 is not reported?
Did my comment https://github.com/intel/llvm/pull/5136#issuecomment-1001938538 start the job?

**Comment by bader** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/617

**Comment by DoyleLi** [comment]
> @tfzhu, @DoyleLi, do you know why validation status for [intel/llvm-test-suite#617](https://github.com/intel/llvm-test-suite/pull/617) is not reported? Did my comment [#5136 (comment)](https://github.com/intel/llvm/pull/5136#issuecomment-1001938538) start the job?

Hi, @bader , the reason for job not been triggered is, there are "/r/n" after the comment body. Please do not append Enter or other character in the body, just type the content and click "comment" button.

**Comment by bader** [comment]
> > @tfzhu, @DoyleLi, do you know why validation status for [intel/llvm-test-suite#617](https://github.com/intel/llvm-test-suite/pull/617) is not reported? Did my comment [#5136 (comment)](https://github.com/intel/llvm/pull/5136#issuecomment-1001938538) start the job?
> 
> Hi, @bader , the reason for job not been triggered is, there are "/r/n" after the comment body. Please do not append Enter or other character in the body, just type the content and click "comment" button.

I suppose these were inserted because I hit "Enter".
Can symbols like "/r", "/t" and "/n" be ignored? These symbols are not printed, so it's hard to tell if there present in my comment or not.

**Comment by DoyleLi** [comment]
> > > @tfzhu, @DoyleLi, do you know why validation status for [intel/llvm-test-suite#617](https://github.com/intel/llvm-test-suite/pull/617) is not reported? Did my comment [#5136 (comment)](https://github.com/intel/llvm/pull/5136#issuecomment-1001938538) start the job?
> > 
> > 
> > Hi, @bader , the reason for job not been triggered is, there are "/r/n" after the comment body. Please do not append Enter or other character in the body, just type the content and click "comment" button.
> 
> I suppose these were inserted because I hit "Enter". Can symbols like "/r", "/t" and "/n" be ignored? These symbols are not printed, so it's hard to tell if there present in my comment or not.

Sure, I will try and add the function to ignore them at the end of comment body.

**Comment by bader** [comment]
@maximdimakov, llvm-test-suite still fails. Please, verify that your patch doesn't break it.
The reason might be outdated branch in llvm-test-suite. If so, please, update it and re-run validation for https://github.com/intel/llvm-test-suite/pull/617.

**Comment by maximdimakov** [comment]
@bader  tests are passed

**Pull request merged**